lib/services/consumed/autocomplete.js

```diff
@@ -0,0 +1,49 @@
+/* @flow */
+import { CompositeDisposable, Disposable } from "atom";
+
+import type { Store } from "../../store";
+
+export const autocompleteProvider = {
+  disposables: new CompositeDisposable(),
+  watchEditor: (editor: atom$TextEditor, labels: Array<string>) => {
+    return;
+  }
+};
+
+export function addWatchEditor(
+  store: Store,
+  watchEditor: atom$AutocompleteWatchEditor
+) {
+  autocompleteProvider.watchEditor = watchEditor;
+  for (let kernel of store.runningKernels) {
+    for (let watch of kernel.watchesStore.watches) {
+      const disposable = watchEditor(this.editor, [
```

Comment:
What does `this.editor` refer to here? I get nervous with `this` outside of a `class`.

Reply:
left from when I ported code. It works weirdly enough, but should be `watch.editor`